src/response/trowtoken.rs

```diff
@@ -0,0 +1,100 @@
+use failure::{Error};
+use std::io::Cursor;
+use std::time::{SystemTime, UNIX_EPOCH};
+use rocket::http::{Status};
+use rocket::http::{ContentType};
+use rocket::request::Request;
+use rocket::response::{Responder, Response};
+use crypto::sha2::Sha256;
+use chrono::Local;
+use jwt::{Header as TokenHeader, Token};
+
+const AUTHORISATION_SECRET: &str = "Bob Marley Rastafaria";
+
+#[derive(Debug, Serialize)]
+pub struct TrowToken;
+
+#[derive(Debug, Serialize, RustcEncodable, RustcDecodable)]
+pub struct BearerToken {
+    user_id: String,
+    client_id: String,
+    scope: String,
+    iat: u64,
+    exp: u64,
+}
+
+fn encode_token() -> Result<String, Error> {
+    // hard coded values for token
+    let username = "admin";
+    let client_id = "docker";
+    let scope = "push/pull";
+    let now = SystemTime::now();
+    let current_time = now.duration_since(UNIX_EPOCH).expect("Time went backwards");
+    let expiry_time = 3600;
+
+    // build token from structure and return token string
+    let token_claims = BearerToken {
+        user_id: username.to_string(),
+        client_id: client_id.to_string(),
+        scope: scope.to_string(),
+        iat: current_time.as_secs(),
+        exp: expiry_time,
+    };
+    let token_header: TokenHeader = Default::default();
+    let bearer_token = Token::new(token_header, token_claims);
+
+    let token_enum = bearer_token.signed(AUTHORISATION_SECRET.as_bytes(), Sha256::new()).ok();
+    let mut token_string = String::new();
+    /*
+    match token_enum {
+        Some(token_enum) => token_string = token_enum,
+        _ => (),
+    };
+     */
+    if let Some(token_enum) = token_enum { token_string = token_enum };
+    Ok(token_string)
```

Comment:
Shouldn't we return an Error rather than Ok and the empty string? Or is there valid cases where it will be empty?